From 8bccea5429a3a7512f64f8cb8b17a47b11d3b68b Mon Sep 17 00:00:00 2001
From: Julian Bouzas <julian.bouzas@collabora.com>
Date: Tue, 29 Oct 2024 15:07:04 -0400
Subject: [PATCH] state-routes: only save route properties if it is available

Some ALSA devices seem to reset the properties of a route when the route becomes
unavailable (Eg when unplugging the headset from the Jack port). This fix avoids
saving the newly reset properties into the default-routes state file when that
happens. This makes sure the old properties are always restored when the route
becomes available again (Eg when plugging the headset back into the Jack port).
---
 src/scripts/device/state-routes.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/scripts/device/state-routes.lua b/src/scripts/device/state-routes.lua
index 1f57c5b7..ba5656a8 100644
--- a/src/scripts/device/state-routes.lua
+++ b/src/scripts/device/state-routes.lua
@@ -232,7 +232,7 @@ store_or_restore_routes_hook = SimpleEventHook {
             Json.Object { index = route_info.index }:to_string ()
         push_select_routes = true
 
-      elseif route.save and route.props then
+      elseif route.save and route.props and route.available ~= "no" then
         -- just save route properties
         log:info (device,
             string.format ("storing route(%s) props of device(%s)",
-- 
2.47.0

